unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuiilderAPI/BuilderAPIResponseResolver.cs

```diff
@@ -13,36 +13,36 @@ public interface IBuilderAPIResponseResolver
         /// </summary>
         /// <param name="result"></param>
         /// <returns></returns>
-        string GetDataFromCall(string result);
-
+        string GetDataFromCall(string result, bool isArray = false);
+        
         /// <summary>
-        ///  Resolve the data from the call removing de OK response and getting only the array
+        /// Resolve the data from the call removing de OK response
         /// </summary>
         /// <param name="result"></param>
         /// <returns></returns>
-        string GetDataFromCallArray(string result);
+        APIResponse GetResponseFromCall(string result, bool isArray = false);
     }
 
     public class BuilderAPIResponseResolver: IBuilderAPIResponseResolver
     {
-        public string GetDataFromCall(string result)
+   
+        public string GetDataFromCall(string result,bool isArray = false)
         {
             JObject jObject = (JObject)JsonConvert.DeserializeObject(result);
             if (jObject["ok"].ToObject<bool>())
             {
+                if (isArray)
+                    return jObject["data"]["items"].ToString();
+                
```

Comment:
Doesn't the `APIResponse` class resolve this?

Reply:
Changed to a new approach that is more clear!